src/main/java/de/tum/in/www1/artemis/web/rest/AthenaResource.java

```diff
@@ -1,81 +1,199 @@
+            AthenaRepositoryExportService athenaRepositoryExportService) {
         this.textExerciseRepository = textExerciseRepository;
         this.textSubmissionRepository = textSubmissionRepository;
+        this.programmingExerciseRepository = programmingExerciseRepository;
+        this.programmingSubmissionRepository = programmingSubmissionRepository;
         this.authCheckService = authCheckService;
+        this.athenaFeedbackSuggestionsService = athenaFeedbackSuggestionsService;
+        this.athenaRepositoryExportService = athenaRepositoryExportService;
     }
 
     /**
-     * GET athena/exercises/:exerciseId/submissions/:submissionId/feedback-suggestions : Get feedback suggestions from Athena
+     * GET athena/text-exercises/:exerciseId/submissions/:submissionId/feedback-suggestions : Get feedback suggestions from Athena for a text exercise
      *
      * @param exerciseId   the id of the exercise the submission belongs to
      * @param submissionId the id of the submission to get feedback suggestions for
      * @return 200 Ok if successful with the corresponding result as body
      */
-    @GetMapping("exercises/{exerciseId}/submissions/{submissionId}/feedback-suggestions")
+    @GetMapping("athena/text-exercises/{exerciseId}/submissions/{submissionId}/feedback-suggestions")
     @EnforceAtLeastTutor
-    public ResponseEntity<List<TextBlockRef>> getFeedbackSuggestions(@PathVariable long exerciseId, @PathVariable long submissionId) {
+    public ResponseEntity<List<TextFeedbackDTO>> getTextFeedbackSuggestions(@PathVariable long exerciseId, @PathVariable long submissionId) {
         log.debug("REST call to get feedback suggestions for exercise {}, submission {}", exerciseId, submissionId);
 
         final var exercise = textExerciseRepository.findByIdElseThrow(exerciseId);
         authCheckService.checkHasAtLeastRoleForExerciseElseThrow(Role.TEACHING_ASSISTANT, exercise, null);
         final var submission = textSubmissionRepository.findByIdElseThrow(submissionId);
         if (submission.getParticipation().getExercise().getId() != exerciseId) {
-            log.error("Exercise id {} does not match submission's exercise id {}", exerciseId, submission.getParticipation().getExercise().getId());
-            throw new ConflictException("Exercise id does not match submission's exercise id", "Exercise", "exerciseIdDoesNotMatch");
+            log.error("Text Exercise id {} does not match submission's exercise id {}", exerciseId, submission.getParticipation().getExercise().getId());
+            throw new ConflictException("Text Exercise id does not match submission's exercise id", "Exercise", "exerciseIdDoesNotMatch");
+        }
+        try {
+            return ResponseEntity.ok(athenaFeedbackSuggestionsService.getTextFeedbackSuggestions(exercise, submission));
+        }
+        catch (NetworkingException e) {
+            return ResponseEntity.status(HttpStatus.SERVICE_UNAVAILABLE).build();
+        }
+    }
```

Comment:
For reviewers: I don't think there's any elegant way to avoid the duplication detected by TeamScale.